.github/workflows/release.yaml

```diff
@@ -0,0 +1,40 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.9, 3.10, 3.11]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          pip install poetry -U
+          poetry install
+
+#      - name: Run tests
+#        run: |
+#          poetry run pytest tests/
+
+      - name: Build and publish
+        if: startsWith(github.ref, 'refs/tags/v')
+        env:
+          POETRY_PYPI_TOKEN_PYPI: ${{ secrets.PYPI_API_TOKEN }}
```

Comment:
this should be added, if you need to pass in a username use __token__ but i don't think you will have to with Poetry